### PR TITLE
(Ready for Review) Start chat with Folder Import

### PR DESCRIPTION
**Pull request by wonderwhy-er**
This is proof of concept for folder import
It allows to start a new chat with selecting a folder
First message in the chat will be message similar to one AI returns with file content to create project from

## Still needs to be tested + other things to do
- There will be problems with binary files like images, need to ignore them for now
- I have ideas of how to support image and other file uploads in future
- May be we need to ask AI to inspect the project and run commands to show preview? 

## Video:
https://github.com/user-attachments/assets/f619d863-6baa-4d53-bf0a-42c020620729

### CODE REVIEW

File: app/components/chat/ImportFolderButton.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import type { Message } from 'ai';
+import { toast } from 'react-toastify';
+
+interface ImportFolderButtonProps {
+  className?: string;
+  importChat?: (description: string, messages: Message[]) => Promise<void>;
+}
+
+const IGNORED_FOLDERS = ['node_modules', '.git', 'dist', 'build', '.next', 'coverage', '.cache', '.vscode', '.idea'];
```
Comment (mrsimpson) [inline app/components/chat/ImportFolderButton.tsx:10]: You may want to include [ignore](https://www.npmjs.com/package/ignore)

You can check that part in my previous PR #289

Comment (wonderwhy-er) [reply app/components/chat/ImportFolderButton.tsx:10]: Good idea, thanks, will add that!

Comment (wonderwhy-er) [reply app/components/chat/ImportFolderButton.tsx]: Just pushed support for it along with ignoring of binnary files.
I have idea of what to do with binary files but now now. In another PR

**Comment by dylanturn** [comment]
Is this something we're going to merge soon? I have a couple projects I'd love to import

**Comment by wonderwhy-er** [comment]
> Is this something we're going to merge soon? I have a couple projects I'd love to import

Yeah, avaiting review and testing from others.
Tested it a bit and will be testing more later

**Comment by mrsimpson** [comment]
What about limits? What happens if – just as an example – you load the projects codebase into it? (Asking for a friend 😬)

Quite sure we're going to see maaaany PRy once this is possible, so maybe get quality assurance in place soon 🔜

**Comment by wonderwhy-er** [comment]
> What about limits? What happens if – just as an example – you load the projects codebase into it? (Asking for a friend 😬)
> 
> Quite sure we're going to see maaaany PRy once this is possible, so maybe get quality assurance in place soon 🔜

By limits you mean how much one can load in to context or chat?
If yes then open source bolt is not ready for that yet.
One of next areas of work is context/token management

And yes, playwright too.

**Comment by coleam00** [comment]
@wonderwhy-er This PR is fantastic - looks awesome and I love how clean your implementation is!

I tested it out myself and it's working perfectly! 

I just have a few suggestions, but honestly I would be good with you merging and improving through another PR if you want just because this feature is so sought after:

1. A loading indicator when importing a folder would be sweet, it can take a while especially for larger uploads.
2. Maybe you could polish up the messaging for the user message that says "Import my files" and the Bolt message that say "Sure, I'll help you import these files"? If you referenced the folder name and made it more complete sentences that would be awesome!
3. When you import a project and then ask for it to continue, it doesn't do an npm install or start the server. Maybe we could add into the project somehow when importing a project to tell it that the next time it does anything it needs to run the necessary commands?
4. I tried importing a large project (46,000 files with node_modules, .next, and .git folders being most of them) and it crashed. I see you set up ignoring for these folders, but maybe that isn't enough because those are all still uploaded to the frontend before being filtered? Not totally sure what the best solution would be for these larger projects.

Nice work man!!

**Comment by coleam00** [comment]
Thanks for your reply here @wonderwhy-er! I look forward to your separate PRs, and what you're saying about Bolt not supporting large projects makes sense. Definitely a larger conversation we have to have as we talked about over in the Think Tank.

**Comment by dhirajbajaj** [comment]
Is this merged? 
Getting Request too large error for 40-mini.  
Its importing .next folder which is increasing the request tokens size.
Is there a way to specify which folders to ignore in UI? 
Or, 
For heavy projects imports can be done in more than 1 request based on token limit sizes

```
responseBody: '{\n' +
      '    "error": {\n' +
      '        "message": "Request too large for gpt-4o-mini in organization org-WKvUC218br4IHWObrEloqFVf on tokens per min (TPM): Limit 200000, Requested 5099836. The input or output tokens must be reduced in order to run successfully. Visit https://platform.openai.com/account/rate-limits to learn more.",\n' +
      '        "type": "tokens",\n' +
      '        "param": null,\n' +
      '        "code": "rate_limit_exceeded"\n' +
      '    }\n' +
      '}\n',
    isRetryable: true,
```

**Pull request merged**